src/Idris/DSL.hs

```diff
@@ -17,13 +17,30 @@ desugar :: SyntaxInfo -> IState -> PTerm -> PTerm
 desugar syn i t = let t' = expandDo (dsl_info syn) t in
                       t' -- addImpl i t'
 
+getNameStr :: Name -> PTerm
+getNameStr (UN nm)   = PConstant $ Str $ str nm
+getNameStr (NS nm _) = getNameStr nm
+getNameStr (MN _ nm) = PConstant $ Str $ str nm
+getNameStr _         = PConstant $ Str "x_unknown"
```

Comment:
It seems to me that this would be much cleaner as a `Maybe String`, rather than picking a special sentinel value here.